### ISSUE TITLE
Game Studio doesnt reload sub projects after changes

### ISSUE THREAD

**Issue by IXLLEGACYIXL** [stride3d/stride#1703]
**Release Type**: Official Release/GitHub (please choose appropriate option)
4.1
**Version**: Version number and/or git branch
Any
**Platform(s)**: Does the problem occur on Windows, Android...?
Game Studio
**Describe the bug**
Game Studio doesnt reload other assemblies besides the main one for refreshing the game studio.
Which means:
You have a second Project, it inherits from a interface and is a Datacontract then Gamestudio will not update it unless you restart it as sub projects are only loaded once.
As far as i could track it down:
https://github.com/stride3d/stride/blob/master/sources/editor/Stride.GameStudio/DebuggingViewModel.cs#L176C1-L181
This never Receives from the Buffer any DLLs besides the main one.

This is caused by
https://github.com/stride3d/stride/blob/master/sources/editor/Stride.Assets.Presentation/AssetEditors/ProjectWatcher.cs#L200
```cs
                if (needProjectReload == false
                    && (e.ChangeType == FileEventChangeType.Deleted || e.ChangeType == FileEventChangeType.Renamed || e.ChangeType == FileEventChangeType.Created)
                    && Path.GetExtension(changedFile)?.ToLowerInvariant() == ".cs"
                    && !UPath.Combine(new UFile(trackedAssembly.Project.FilePath).GetFullDirectory(), new UDirectory("obj")).Contains(new UFile(changedFile)))
                {
                    needProjectReload = true;
                }
```
this will be always true for the main project, so it will trigger the reload of the project here
https://github.com/stride3d/stride/blob/master/sources/editor/Stride.Assets.Presentation/AssetEditors/ProjectWatcher.cs#L229
So it will never reach the other dlls

Also swapping the order of the dlls wont help as it will create a endless loop
1. second project rebuilds
2. main project changes so it gets build
3. second project rebuilds
4. ...

**Expected behavior**
It should behave like in the main project where i add a type in code and it properly reloads and i dont have to restart game studio

**Additional context**
The Assembly Registry and TypeDescriptorExtension work.
I managed to get it "working" by skipping hardcoded dlls to check if the other stuff works and it does.
the skip caused 3 times reloading the dlls as:
1. main project changed and gets rebuild
2. second project changed and gets rebuild
3. both projects get rebuild ( which would be expected )

**Comment by IXLLEGACYIXL** [comment]
the issue was that the reload gets triggered by the main assembly and it will be the first one that gets triggered, so it will always only reload the main project and not the subprojects

now the trigger fires off when the dll fits to the change so it triggers the reload and not always the main project

**Comment by IXLLEGACYIXL** [comment]
fixed with #1704
